### PR TITLE
docs: Send Notification to teams on Exit

### PR DESCRIPTION
**Pull request by iflan7744**
Signed-off-by: Mohamed Iflan Zameer <iflan_mohamed@yahoo.com>

This will send notifications to Microsoft teams on exit.

### CODE REVIEW

File: examples/exit-handler-teams.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# In order to configure, create teams webhook url for it.
+# The documentation to do so is here: https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook
+# Replace TEAMS_WEBHOOK_URL_HERE on line 31 with that webhook url
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: exit-handler-slack-
+spec:
+  entrypoint: say-hello
+  onExit: exit-handler
```
Comment (sarabala1979) [inline examples/exit-handler-teams.yaml:11]: can you  update with lifecyclehook?

**Review by sarabala1979 — CHANGES_REQUESTED** [review]

**Review by terrytangyuan — COMMENTED** [review]
We already have examples for exit handlers.

**Comment by sarabala1979** [comment]
@terrytangyuan right, it is just an exit handler. Do you want it to add to https://argoproj-labs.github.io/argo-workflows-catalog/? if you think this will be unique for teams.

**Comment by iflan7744** [comment]
> We already have examples for exit handlers.

Yeah, That's example is for Slack and email. My PR will send notification to Microsoft team based on lifecycle hook (EXIT)

File: examples/exit-handler-teams.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# This example shows how to hit a teams webhook on an exit handler.
```
Comment (caelan-io) [inline examples/exit-handler-teams.yaml:1]: Suggested rephrase: "This example shows how to send a Microsoft Teams notification with an exit handler by hitting a Teams webhook." 

also, "... create a Microsoft Teams webhook url for it."

File: examples/exit-handler-teams.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# In order to configure, create teams webhook url for it.
+# The documentation to do so is here: https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook
+# Replace TEAMS_WEBHOOK_URL_HERE on line 31 with that webhook url
```
Comment (caelan-io) [inline examples/exit-handler-teams.yaml:4]: Small QC: I'm seeing this come through on line 33 of the file, not line 31. Update? 

Smart to call this out, though 👍

**Review by caelan-io — COMMENTED** [review]
Nice example! I would rephrase as "Microsoft Teams" and "notification" throughout these comments to help users who are searching in our docs or on Google/GitHub find it faster with the following search terms:
- microsoft teams notification argo workflows
- teams notification argo workflows
- how to send teams notification argo

This is only a suggestion based on my experience optimizing docs for search behavior.

**Comment by caelan-io** [comment]
Also, I would support adding this to the repo docs based on increase in Teams adoption over the past couple years. 👍

File: examples/exit-handler-teams.yaml (hunk 2 above)
Comment (iflan7744) [reply examples/exit-handler-teams.yaml]: @caelan-io done

File: examples/exit-handler-teams.yaml (hunk 1 above)
Comment (caelan-io) [reply examples/exit-handler-teams.yaml]: @sarabala1979 looks like this was addressed. can we merge to provide this example to the community?

File: examples/exit-handler-teams.yaml (hunk 2 above)
Comment (caelan-io) [reply examples/exit-handler-teams.yaml]: awesome, thanks!

File: examples/exit-handler-teams.yaml (hunk 1 above)
Comment (caelan-io) [reply examples/exit-handler-teams.yaml]: bump: let's get this doc merged 👍

**Review by caelan-io — APPROVED** [review]
LGTM

**Review by alexec — DISMISSED** [review]

**Comment by alexec** [comment]
Run codegen pls
